DM-25028 change butler create --config-file to --seed-config & DM-25151 in butler CLI add -h as an alias for --help

n8pease opened the pull request.

n8pease committed:
change config-file default help

n8pease committed:
change create cmd opt 'config-file' to 'seed-config'

n8pease committed:
make a command test base

n8pease committed:
convert command tests to common base

n8pease committed:
add '-h' as an alias for '--help'

timj commented on python/lsst/daf/butler/cli/butler.py:
```diff
@@ -253,7 +253,7 @@ def get_command(self, context, name):
 
 
-@click.command(cls=LoaderCLI)
+@click.command(cls=LoaderCLI, context_settings=dict(help_option_names=['-h', '--help']))
```
General comment but in daf_butler we always prefer double quotes to single quotes.

n8pease replied:
yeah. This was copy/paste and I didn't notice the quotes, will fix.

timj commented on python/lsst/daf/butler/cli/cmd/commands.py:
```diff
@@ -50,7 +49,7 @@ def butler_import(*args, **kwargs):
+@click.option("--seed-config", help="Path to an existing YAML config file to apply (on top of defaults).")
 @click.option("--standalone", is_flag=True, help="Include all defaults in the config file in the repo, "
               "insulating the repo from changes in package defaults.")
 @click.option("--override", "-o", is_flag=True, help="Allow values in the supplied config to override any "
```
We probably want to remove the `-o` here since it's a very rare option and it's likely that we'll be using `-o` elsewhere.

timj replied:
Also change text to "...supplied seed config to override all repo defaults"

timj commented on python/lsst/daf/butler/tests/mockeredTest.py:
```diff
@@ -0,0 +1,104 @@
+from ..cli import butler
+
+
+class MockeredTestBase(unittest.TestCase, abc.ABC):
```
"mockered"?

n8pease replied:
at the risk of being too cute, I have a class called Mocker that wraps a static instance of MagicMock. This test uses a Mocker and so is Mockered.

timj reviewed (approved):
Nice refactoring of the tests.

n8pease committed:
adjust override option parameters

remove -o because --option is a rarely used option and so don't
squat on that short option alias.

fix the --override help text
